sidechain/contracts/group/Group.sol

```diff
@@ -312,26 +335,174 @@ contract Group is GroupI, UsingExternalStorage
         return keccak256(abi.encodePacked(_groupId, _role, _secretHash, _nonce));
```

Comment:
Add the function name in the hash just to guarantee signature uniqueness like you did in `prepareCreateGroup`
```
return keccak256(
    abi.encodePacked(
        address(this), 
        "createGroup",
        _metadataLocator,
        _nonce 
    )
);
```

Edge case I know, but it could happen that we have two different functions with identical arguments and consequently the same signature would be valid for both functions.

Example:
```
function upgradePlayer(uint _playerId) return bool;
function removePlayer(uint _playerId) return bool;
```
If you send a meta-tx (signature) for `upgradePlayer`. The relayer or a man in the middle (pool watcher) could intercept the transaction and send the same signature to `removePlayer`.

Reply:
resolved